hrms/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -1494,11 +1585,14 @@ def get_taxable_earnings(self, allow_tax_exemption=False, based_on_payment_days=
 			}
 		)
 
-	def get_future_recurring_period(self, additional_salary, relieving_date=None):
-		to_date = frappe.db.get_value("Additional Salary", additional_salary, "to_date")
+	def get_future_recurring_period(
+		self,
+		additional_salary,
+	):
+		to_date = frappe.db.get_value("Additional Salary", additional_salary, "to_date", cache=True)
 
-		if relieving_date:
-			to_date = relieving_date
+		if self.relieving_date:
+			to_date = self.relieving_date
```

Comment:
Maybe we can avoid a db call here since if relieving date is set, `to_date` gets overwritten anyway.